contracts/dapis/proxies/DapiProxy.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IDapiProxy.sol";
+
+contract DapiProxy is IDapiProxy {
+    address public immutable dapiServer;
+    bytes32 public immutable dapiNameHash;
+
+    constructor(address _dapiServer, bytes32 _dapiName) {
+        dapiServer = _dapiServer;
+        dapiNameHash = keccak256(abi.encodePacked(_dapiName));
```

Comment:
Should we emit events in the constructor for apps to collect them (operations-database handler maybe)?

Reply:
@Siegrift mentioned that such events can be spoofed (a contract can emit them in a loop, etc.) and be annoying. We need the apps to also know the metadata emitted by the factory contract anyway, so it's better to also emit these at the factory contract and only listen for events from that contract.